Tracks/reapertips_Track-Icon-Selector.lua

```diff
@@ -0,0 +1,539 @@
+--   - Dockable Window: Conveniently dock the script window to the left of your tracks for easy access.
+--   - Yellow Outline Around Used Icons: Instantly identify tracks that already have icons applied.
+--
+--   ## Creating and Modifying Categories:
+--
+--   To create or modify categories for your icons, navigate to the following REAPER folder:
+--
+--   REAPER Folder > Data > track_icons
+--   Within this folder, create or modify folders to represent your desired categories. The script will automatically recognize these folders as categories in the sidebar, allowing you to filter icons based on their assigned categories.
+
+-- @description Track Icon Selector
+-- @author Sexan
+-- @license GPL v3
+-- @version 1.0
+-- @changelog
+--  Initial release
+
+local r                = reaper
+package.path           = r.ImGui_GetBuiltinPath() .. '/?.lua'
+local os_separator     = package.config:sub(1, 1)
+
+local imgui            = require "imgui" "0.9.1"
+local reaper_path      = r.GetResourcePath()
+
+local function NeedDeps()
+    local deps = {}
+
+    if not r.ImGui_GetVersion then
+        deps[#deps + 1] = '"Dear Imgui"'
+    end
+    if #deps ~= 0 then
+        r.ShowMessageBox("Script needs ReaImGui.\nPlease Install it in next window", "MISSING DEPENDENCIES", 0)
+        r.ReaPack_BrowsePackages(table.concat(deps, " OR "))
+        return true
+    end
+end
+if NeedDeps() then return end
+
+local COLORS           = {
+    ["win_bg"] = 0x282828ff,
+    ["sidebar_col"] = 0x2c2c2cff,
+    ["hover_col"] = 0x3c6191ff,
+    ["outline_col"] = 0xffcb40ff,
+    ["text_active"] = 0xf1f2f2ff,
+    ["text_inactive"] = 0x999B9Fff
+}
+
+local ALWAYS_SHOW_CATEGORIES = false
+local QUIT_ON_SELECT         = false
+local TOOLTIPS               = true
+local SIDE_WIDTH = 100
+
+function StringToTable(str)
+    local f, err = load("return " .. str)
+    return f ~= nil and f() or nil
+end
+
+if r.HasExtState("TRACK_ICONS4", "STORED_DATA") then
+    local stored = r.GetExtState("TRACK_ICONS4", "STORED_DATA")
+    if stored ~= nil then
+        local storedTable = StringToTable(stored)
+        if storedTable ~= nil then
+            COLORS["win_bg"] = storedTable.win_col
+            COLORS["sidebar_col"] = storedTable.sidebar_col
+            COLORS["hover_col"] = storedTable.hover_col
+            COLORS["text_active"] = storedTable.text_active
+            COLORS["text_inactive"] = storedTable.text_inactive
+            COLORS["outline_col"] = storedTable.outline_col
+            ALWAYS_SHOW_CATEGORIES = storedTable.ALWAYS_SHOW_CATEGORIES
+            QUIT_ON_SELECT = storedTable.QUIT_ON_SELECT
+            TOOLTIPS = storedTable.TOOLTIPS
+            SIDE_WIDTH = storedTable.SIDE_WIDTH
+        end
+    end
+end
+
+local OPEN_CATEGORIES = ALWAYS_SHOW_CATEGORIES
+
+local MAIN_PNG_TBL = {
+    [0] = {}
+}
+local FILTERED_PNG
+
+local icon_size = 32
+
+r.set_action_options(1)
+
+local ctx = imgui.CreateContext('Track Icons', imgui.ConfigFlags_DockingEnable)
```

Comment:
`DockingEnable` overrides the user's preference. Is this the desired behavior? Docking is enabled by default.